src/main/scala/mesosphere/marathon/api/akkahttp/v2/TasksController.scala

```diff
@@ -0,0 +1,105 @@
+package mesosphere.marathon
+package api.akkahttp.v2
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.server.Route
+import com.typesafe.scalalogging.StrictLogging
+import mesosphere.marathon.api.akkahttp.Controller
+import mesosphere.marathon.api.akkahttp.Directives.pathEndOrSingleSlash
+import mesosphere.marathon.core.appinfo.EnrichedTask
+import mesosphere.marathon.core.condition.Condition
+import mesosphere.marathon.core.group.GroupManager
+import mesosphere.marathon.core.health.{ Health, HealthCheckManager }
+import mesosphere.marathon.core.instance.Instance
+import mesosphere.marathon.core.instance.Instance.Id
+import mesosphere.marathon.core.task.tracker.InstanceTracker
+import mesosphere.marathon.plugin.auth.{ Authenticator, _ }
+import play.api.libs.json.Json
+import mesosphere.marathon.raml.{ AnyToRaml, EnrichedTasksList, Writes }
+import mesosphere.marathon.state.PathId
+
+import scala.async.Async._
+import scala.concurrent.{ ExecutionContext, Future }
+
+class TasksController(instanceTracker: InstanceTracker, groupManager: GroupManager, healthCheckManager: HealthCheckManager)(
+    implicit
+    val actorSystem: ActorSystem,
+    val executionContext: ExecutionContext,
+    val authenticator: Authenticator,
+    val authorizer: Authorizer
+) extends Controller with StrictLogging {
+
+  import mesosphere.marathon.api.akkahttp.Directives._
+  import mesosphere.marathon.api.akkahttp.EntityMarshallers._
+  import mesosphere.marathon.raml.EnrichedTaskConversion._
+
+  override val route = {
+    get {
+      pathEndOrSingleSlash {
+        listTasks()
+      }
+    }
+  }
+
+  private def listTasks(): Route = {
+    authenticated.apply { implicit identity =>
+      authorized(ViewResource, AuthorizedResource.SystemConfig).apply {
+        parameters("status".?, "status[]".as[String].*) { (statusParameter, statusesParameter) =>
```

Comment:
Hm, should `statusesParameter` rather be `statusParameters`? I'm actually surprised that we need `"status".?` at all. Is this not covered by `"status[]".as[String].*`?

Reply:
I think this was already answered in one of your other comments - to keep backward compatibility, I need to accept "status" and "status[]"